src/Shouldly.Tests/ShouldBeEquivalentTo/ObjectScenario.cs

```diff
@@ -285,5 +286,17 @@ public void ShouldPassWhenObjectContainsInfiniteLoop()
 
             subject.ShouldBeEquivalentTo(expected);
         }
+
+        [Fact]
+        public void ShouldThrowSensibleErrorWhenIndexersUsed()
+        {
+            var subject = new IndexableObject(new List<string>{"foo", "bar"});
+            var expected = new IndexableObject(new List<string>{"a", "b"});
+
+            Verify.ShouldFail(() =>
+                    subject.ShouldBeEquivalentTo(expected),
+                    errorWithSource: "Comparing indexer property types is not supported",
+                    errorWithoutSource:"Comparing indexer property types is not supported");
```

Comment:
Normally I'd rather think of this as not being a test failure but being a Shouldly API usage failure and throw `NotSupportedException`. What do you think?

Also just to nitpick, what do you think about `Comparing types that have indexers is not supported.`? I like having full sentences including end marks for user-facing messages coming out of a library, including XML docs and exception messages.